spki/src/traits.rs

```diff
@@ -93,3 +96,11 @@ pub trait EncodePublicKey {
         Ok(doc.write_pem_file(path, SubjectPublicKeyInfoRef::PEM_LABEL, line_ending)?)
     }
 }
+
+/// Returns AlgorithmIndentifier associated with the structure. This is mostly useful for Signing
+/// and Validation keys.
+#[cfg(feature = "alloc")]
+pub trait AssociatedAlgorithmIdentifier {
```

Comment:
As a more specific recommendation:

```suggestion
pub trait DynAssociatedAlgorithmIdentifier {
```

...leaving `AssociatedAlgorithmIdentifier` for a trait which uses an associated constant.

Reply:
ack, pushing an updated version